dlt/extract/utils.py

```diff
@@ -119,6 +119,29 @@ def check_compat_transformer(name: str, f: AnyFun, sig: inspect.Signature) -> in
     return meta_arg
 
 
+def wrap_async_generator(gen: Any) -> Any:
+    """Wraps an async generator into a list of awaitables"""
+    is_running = False
+    exhausted = False
+
+    async def run() -> TDataItems:
+        nonlocal is_running, exhausted
+        try:
+            return await gen.__anext__()
+        except StopAsyncIteration:
+            exhausted = True
+            raise
+        finally:
+            is_running = False
+
+    # it is best to use the round robin strategy here if multiple async generators are used in resources
+    while not exhausted:
```

Comment:
I think we should use the same strategy that user wants? anyway. IMO if you have two async generators they will be executed in parallel in futures, right?

Reply:
not if we emit a list of awaitables, then one list will be processed first, then the other (you can see that in the test)